### PR TITLE
Upgrade Python imports for Python 3.10 compatibility

### LINKED ISSUE #18: Upgrade imported packages for python 3.10 compatibility
Pytest 6.2.5 has issues with python 3.10.  While there, fix a few test issues & upgrade other packages.

### PR DESCRIPTION
**Pull request by fhcarter**
Fixes #18 

Upgrade imports to more recent versions (generally most recent).  Pytest 6.x has compatibility issues with Python 3.10.  This should solve those.

While there, fix a few test issues having to do with Vantiq changes over time.

Changes to the python ext source sdk & connectors will follow -- same types of changes -- as will publishing this version to Pypi.

### CODE REVIEW

File: src/test/python/test_VantiqSDKLive.py [hunk 1]
```diff
@@ -169,19 +170,27 @@ def check_callback_topic_publish_saveit(self, what: str, msg: dict):
...
+            print('Not saving message: looking for', self.looking_for, ', found: ', what)
 
     async def subscriber_callback(self, what: str, details: dict) -> None:
         print('Subscriber got a callback -- what: {0}, details: {1}'.format(what, details))
         self.callback_count += 1
         self.callbacks.append(what)
         if self.message_checker:
             self.message_checker(what, details)
+        else:
+            print('Np message checker in place for ', what, '::', details)
```
Comment (skyler1ackerman) [inline src/test/python/test_VantiqSDKLive.py:187]:
```suggestion
            print('No message checker in place for ', what, '::', details)
```

File: src/test/python/test_VantiqSDKMock.py [hunk 2]
```diff
@@ -532,18 +532,18 @@ async def check_crud_operations(self, mocked, client: Vantiq, skip_pretest_clean
...
         assert isinstance(vr, VantiqResponse)
         assert not vr.is_success
         errs = vr.errors
         assert isinstance(errs, list)
         ve = errs[0]
-        assert ve.message == "The requested instance ('[name:foo]') of the k8sclusters resource could not be found."
+        assert ve.message == "The requested instance ('{name=foo}') of the k8sclusters resource could not be found."
         assert ve.code == 'io.vantiq.resource.not.found'
-        assert ve.params == ['k8sclusters', '[name:foo]']
+        assert ve.params == ['k8sclusters', '{name=foo}']
```
Comment (skyler1ackerman) [inline src/test/python/test_VantiqSDKMock.py:546]: Is this a Vantiq formatting change?

**Review by skyler1ackerman — APPROVED** [review]
LGTM, mostly dependency updates. Just had a question about the bracket changes in test_VantiqSDKMock.py

**Review by steve-langley — APPROVED** [review]
LGTM - seems reasonable, but that's probably not worth much.

File: src/test/python/test_VantiqSDKMock.py (hunk 2 above)
Comment (fhcarter) [reply src/test/python/test_VantiqSDKMock.py:546]: yes, looks like formatting changed -- these allow the tests to pass when run against a real server.

**Pull request merged**